internal/resource/route.go

```diff
@@ -638,6 +684,22 @@ func init() {
 					},
 				},
 			},
+			{
+				Title: ExpressionRouteRuleTitle,
+				Description: "When 'expression' is defined, 'priority' is required and " +
+					"'snis', 'sources' or 'destinations' cannot be set.",
+				If: &generator.Schema{
+					Required: []string{"expression"},
+				},
+				Then: &generator.Schema{
+					Required: []string{"priority"},
+					Properties: map[string]*generator.Schema{
+						"snis":         {Not: &generator.Schema{}},
+						"sources":      {Not: &generator.Schema{}},
+						"destinations": {Not: &generator.Schema{}},
```

Comment:
Other than these, the following fields cannot be set with `expression`: `paths`, `hosts`, `headers`, `methods`, `regex_priority`, `path_handling`

Reply:
done.  had to make the `r.ProcessDefaults()` method deal differently for expression routes.

Reply:
also made the messages more specific, otherwise it was too long to be helpful.

Reply:
Yeah that looks much better now, thanks!